lock/src/main/java/com/auth0/android/lock/LockActivity.java

```diff
@@ -342,6 +351,43 @@ public void onDatabaseAuthenticationRequest(DatabaseChangePasswordEvent event) {
         request.start(changePwdCallback);
     }
 
+    @SuppressWarnings("unused")
+    @Subscribe
+    public void onEnterpriseAuthenticationRequest(EnterpriseWebLoginEvent event) {
+        //same as social authentication
+        if (options == null) {
+            return;
+        }
+
+        progress.showProgress();
+        String pkgName = getApplicationContext().getPackageName();
+        CallbackHelper helper = new CallbackHelper(pkgName);
+        lastIdp = new WebIdentityProvider(helper, options.getAccount(), idpCallback);
+        lastIdp.setUseBrowser(options.useBrowser());
+        lastIdp.setParameters(options.getAuthenticationParameters());
+        lastIdp.start(LockActivity.this, event.getConnectionName());
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe
+    public void onEnterpriseAuthenticationRequest(EnterpriseROLoginEvent event) {
+        //almost the same as database authentication
+        if (options == null) {
+            return;
+        } else if (event.getConnectionName().equals(Strategies.ActiveDirectory.getName()) && configuration.getDefaultActiveDirectoryConnection() == null) {
+            return;
+        } else if (configuration.getEnterpriseStrategies().isEmpty()) {
+            return;
+        }
+
+        progress.showProgress();
+        AuthenticationAPIClient apiClient = new AuthenticationAPIClient(options.getAccount());
+        apiClient.setDefaultDbConnection(event.getConnectionName());
```

Comment:
We should not do this, we should instead set the `connection` entry in the parameters of the request.